rdr_service/genomic/genomic_job_components.py

```diff
@@ -1694,6 +1697,63 @@ def calculate_contamination_category(self, sample_id, raw_contamination, member:
 
         return contamination_category
 
+    def _set_metrics_data_file_paths(self, row: dict):
+        cvl_site_bucket_map = config.getSettingJson(config.GENOMIC_CVL_SITE_BUCKET_MAP)
+        prefix_map = config.getSettingJson(config.GENOMIC_CVL_SITE_PREFIX_MAP)
+        site_id = self.file_obj.fileName.split('_')[0].lower()
+        cvl_bucket = cvl_site_bucket_map[site_id]
+        gc_bucket = config.getSetting(cvl_bucket)
+
+        if row['genometype'] == 'aou_array':
+            for file_def in array_file_types_attributes:
+                if file_def['required']:
+                    if 'idat' in file_def["file_type"]:
+                        file_path = f'gs://{gc_bucket}/Genotyping_sample_raw_data/{row["chipwellbarcode"]}' + \
+                                    f'_{file_def["file_type"]}'
+                    else:
+                        file_path = f'gs://{gc_bucket}/Genotyping_sample_raw_data/{row["chipwellbarcode"]}.' + \
+                                    f'{file_def["file_type"]}'
+                    row[file_def['file_path_attribute']] = file_path
+                    row[file_def['file_received_attribute']] = 1
+            # Need to set wgs file received flags to 0 to prevent an error when ingesting updated AW2
+            for file_def in wgs_file_types_attributes:
+                if file_def['required']:
+                    row[file_def['file_received_attribute']] = 0
+                elif 'gvcf' in file_def['file_type']:
+                    row[file_def['file_received_attribute']] = 0
+
+        elif row['genometype'] == 'aou_wgs':
+            for file_def in wgs_file_types_attributes:
+                if file_def['required']:
+                    file_path = f'gs://{gc_bucket}/{prefix_map[site_id][file_def["file_type"]]}/{site_id.upper()}_' + \
+                                f'{row["biobankid"]}_{row["sampleid"]}_{row["limsid"]}_1.{file_def["file_type"]}'
+                    row[file_def['file_path_attribute']] = file_path
+                    row[file_def['file_received_attribute']] = 1
+                # gvcf files not required but needed for AW3 generation
+                elif 'gvcf' in file_def['file_type']:
+                    file_path = f'gs://{gc_bucket}/{prefix_map[site_id][file_def["file_type"]]}/{site_id.upper()}_' + \
+                                f'{row["biobankid"]}_{row["sampleid"]}_{row["limsid"]}_1.{file_def["file_type"]}'
+                    row[file_def['file_path_attribute']] = file_path
+                    row[file_def['file_received_attribute']] = 1
+            # Need to set array file received flags to 0 to prevent an error when ingesting updated AW2
+            for file_def in array_file_types_attributes:
+                if file_def['required']:
+                    row[file_def['file_received_attribute']] = 0
+
+        return row
```

Comment:
Think it might be a good idea to refactor this to make it a little bit cleaner in the execution, seems like some of this code, mainly setting the attributes is being repeated muiltiple times.

Reply:
I concur. Since an AW2 will either be for Array or WGS, you can reduce the complexity by splitting these up into separate functions. 
You can also go ahead and make the gVCF "required" in the file_def dict, since we are now getting rid of the reconciliation process and only need these paths for the AW3. Doing so will reduce the need for a separate condition for the gVCF, since it is essentially required anyway.
All `file_received_attributes` for the AW2's genome type should be set when creating the gc_validation_metrics record. We wouldn't ever ingest an Array AW2 for a WGS sample, and vice-versa. They will always have different sample IDs, so you should be able to remove setting these attributes to 0 from `_set_metrics_data_file_paths()`. I'm not sure what the error was, but let me know if there are any questions about the above.